pkg/scheduler/plugins/deviceshare/plugin.go

```diff
@@ -293,7 +293,9 @@ func (p *Plugin) Filter(ctx context.Context, cycleState *framework.CycleState, p
 
 	nodeDeviceInfo := p.nodeDeviceCache.getNodeDevice(node.Name, false)
 	if nodeDeviceInfo == nil {
-		return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrMissingDevice)
+		state.skip = true
```

Comment:
And the lifetime of the state is one scheduling cycle, so cannot fix as this.

Reply:
Yes, it will work for the entire scheduling cycle, and I'm handling it this way in the hope that the plugin wouldn't handle nodes without devices.

so, if there is no device it will only be affected by the result of noderesourcefit, and if there is a device it will be handled by deviceshare

Reply:
But state.skip means that Pod does not need to request GPU or other devices.